### DISCUSSION TITLE
Quick start issues

### DISCUSSION THREAD

**Discussion by hlallo-mind** [wgu-opensource/osmt#185] [Asking for Help] [answered]
Anyone seeing this error when trying to install the quick start. 

java.util.concurrent.ExecutionException: java.net.UnknownHostException: osmt_quickstart_elasticsearch_1: Name or service not known
osmt_quickstart-app-1            | 	at org.apache.http.concurrent.BasicFuture.getResult(BasicFuture.java:71) ~[httpcore-4.4.14.jar!/:4.4.14]
osmt_quickstart-app-1            | 	at org.apache.http.concurrent.BasicFuture.get(BasicFuture.java:84) ~[httpcore-4.4.14.jar!/:4.4.14]
osmt_quickstart-app-1            | 	at org.apache.http.impl.nio.client.FutureWrapper.get(FutureWrapper.java:70) ~[httpasyncclient-4.1.4.jar!/:4.1.4]
osmt_quickstart-app-1            | 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:255) ~[elasticsearch-rest-client-7.9.2.jar!/:7.9.2]
osmt_quickstart-app-1            | 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:246) ~[elasticsearch-rest-client-7.9.2.jar!/:7.9.2]
osmt_quickstart-app-1            | 	at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1613) ~[elasticsearch-rest-high-level-client-7.9.2.jar!/:7.9.2]
osmt_quickstart-app-1            | 	at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1598) ~[elasticsearch-rest-high-level-client-7.9.2.jar!/:7.9.2]
osmt_quickstart-app-1            | 	at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1565) ~[elasticsearch-rest-high-level-client-7.9.2.jar!/:7.9.2]
osmt_quickstart-app-1            | 	at org.elasticsearch.client.RestHighLevelClient.info(RestHighLevelClient.java:766) ~[elasticsearch-rest-high-level-client-7.9.2.jar!/:7.9.2]
osmt_quickstart-app-1            | 	at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.lambda$getClusterVersion$17(ElasticsearchRestTemplate.java:413) ~[spring-data-elasticsearch-4.2.4.jar!/:4.2.4]
osmt_quickstart-app-1            | 	at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.execute(ElasticsearchRestTemplate.java:383) ~[spring-data-elasticsearch-4.2.4.jar!/:4.2.4]
osmt_quickstart-app-1            | 	at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.getClusterVersion(ElasticsearchRestTemplate.java:413) ~[spring-data-elasticsearch-4.2.4.jar!/:4.2.4]
osmt_quickstart-app-1            | 	at org.springframework.data.elasticsearch.core.AbstractElasticsearchTemplate.initialize(AbstractElasticsearchTemplate.java:104) ~[spring-data-elasticsearch-4.2.4.jar!/:4.2.4]
osmt_quickstart-app-1            | 	at org.springframework.data.elasticsearch.core.ElasticsearchRestTemplate.<init>(ElasticsearchRestTemplate.java:109) ~[spring-data-elasticsearch-4.2.4.jar!/:4.2.4]
osmt_quickstart-app-1            | 	at edu.wgu.osmt.elasticsearch.ElasticsearchClientManager.elasticsearchTemplate(ElasticsearchClientManager.kt:34) ~[classes!/:1.1.0]
osmt_quickstart-app-1            | 	at edu.wgu.osmt.elasticsearch.ElasticsearchClientManager$$EnhancerBySpringCGLIB$$cafd7b73.CGLIB$elasticsearchTemplate$4(<generated>) ~[classes!/:1.1.0]
osmt_quickstart-app-1            | 	at edu.wgu.osmt.elasticsearch.ElasticsearchClientManager$$EnhancerBySpringCGLIB$$cafd7b73$$FastClassBySpringCGLIB$$c2528524.invoke(<generated>) ~[classes!/:1.1.0]
osmt_quickstart-app-1            | 	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.3.10.jar!/:5.3.10]
osmt_quickstart-app-1            | 	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331) ~[spring-context-5.3.10.jar!/:5.3.10]

**Comment by JohnKallies** [comment]
Hello,

Would you please paste the output of these commands in a reply? I don't foresee anything sensitive or private in that command output.

`./osmt_cls.sh -v`

`docker ps` (with your Quickstart instance started, even broken)

Hopefully we can get you moving forward. Thanks!

**Comment by JohnKallies** [comment]
Also, I'm moving this from an Issue to a Discussion, and if we identify an actionable issue, we can then create one :)

**Reply by hlallo-mind** [reply]
Thank you!

**Comment by hlallo-mind** [comment]
___  ___ __  __ _____   ___ _    ___
  / _ \/ __|  \/  |_   _| / __| |  |_ _|
 | (_) \__ \ |\/| | | |  | (__| |__ | |
  \___/|___/_|  |_| |_|   \___|____|___|

INFO:  Validating software and SDKs available for OSMT on this machine...
INFO:  Reporting operating system information...
Darwin MindEdges-MacBook-Pro.local 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64 x86_64

INFO:  Checking git...
INFO:  Git detected at /usr/bin/git
git version 2.32.0 (Apple Git-132)

INFO:  Checking Docker...
INFO:  Docker detected at /usr/local/bin/docker
INFO:  Checking Docker version...
Client:
 Cloud integration: v1.0.22
 Version:           20.10.12
 API version:       1.41
 Go version:        go1.16.12
 Git commit:        e91ed57
 Built:             Mon Dec 13 11:46:56 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.5.0 (74594)
 Engine:
  Version:          20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.12
  Git commit:       459d0df
  Built:            Mon Dec 13 11:43:56 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
INFO:  Docker engine version 20.10.12 is >= 17.6.0

INFO:  Checking Java...
INFO:  Java detected at /usr/bin/java
INFO:  Checking Java JDK for version 11 or greater...
INFO:  Java version 12.0.2 is >= 11.0.0

INFO:  Maven version: Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /usr/local/Cellar/maven/3.8.4/libexec
Java version: 17.0.2, vendor: Homebrew, runtime: /usr/local/Cellar/openjdk/17.0.2/libexec/openjdk.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.2.1", arch: "x86_64", family: "mac"

INFO:  OSMT development recommends NodeJS version v16.13.0 or greater. Maven uses an embedded copy of NodeJS v16.13.0.
INFO:  NodeJS version: v16.3.0

INFO:  OSMT development recommends npm version 8.1.0 or greater. Maven uses an embedded copy of npm 8.1.0.
INFO:  npm version: 7.15.1

INFO:  Checking environment files used in local OSMT instances...

INFO:  Checking /Users/mindedge/Documents/mind/osmt/osmt-quickstart.env for invalid default values (xxxxxx)
INFO:  No invalid default values in /Users/mindedge/Documents/mind/osmt/osmt-quickstart.env

INFO:  Checking /Users/mindedge/Documents/mind/osmt/api/osmt-dev-stack.env for invalid default values (xxxxxx)
INFO:  No invalid default values in /Users/mindedge/Documents/mind/osmt/api/osmt-dev-stack.env

CONTAINER ID   IMAGE                                                  COMMAND                  CREATED             STATUS              PORTS                                            NAMES
36795bd7a65b   wguopensource/osmt-app:latest                          "/opt/osmt/bin/docke…"   About an hour ago   Up About a minute   0.0.0.0:8080->8080/tcp                           osmt_quickstart-app-1
6981a7e95b13   docker.elastic.co/elasticsearch/elasticsearch:7.16.2   "/bin/tini -- /usr/l…"   About an hour ago   Up About a minute   0.0.0.0:9200->9200/tcp, 0.0.0.0:9300->9300/tcp   osmt_quickstart-elasticsearch-1
e62c590fd0ad   mysql:8.0                                              "docker-entrypoint.s…"   About an hour ago   Up About a minute   0.0.0.0:3306->3306/tcp, 33060/tcp                osmt_quickstart-db-1
9ae6f6d448c2   redis:6.0.6

**Comment by JohnKallies** [comment, marked as answer]
Thanks for replying.

Based on your `docker ps` output, your version of Docker is using hyphens as a separator. Some versions use an underscore. See this post for more context -- https://github.com/wgu-opensource/osmt/discussions/146

tldr; modify the Docker service host names in your env files to match the container names in your `docker ps` output.

**Reply by hlallo-mind** [reply]
Thank you that seems to help. Its importing items now. I will wait to see if it fully starts up before i ask to close if possible.

**Comment by hlallo-mind** [comment]
@JohnKallies We are good to close this discussion, I am all set

**Reply by JohnKallies** [reply]
That's good news. Can I ask what you learned? Is it something the rest of us could benefit from?
